aten/src/ATen/native/quantized/cpu/qhardsigmoid.cpp

```diff
@@ -87,4 +87,10 @@ Tensor hardsigmoid_quantized_cpu(const Tensor& qx) {
   return qy;
 }
 
+Tensor& hardsigmoid_out_quantized_cpu(const Tensor& qx, Tensor& result) {
+  // TODO(future PR): optimize the qnnpack implementation, if needed
+  result = hardsigmoid_quantized_cpu(qx);
```

Comment:
This is not really how `out` variants work. You should be re-using the given `result` Tensor, re-sizing it if necessary and then inplace modifying it with the values you need.

Reply:
sorry for putting up a too-hacky PR, will re-do